libbeat/monitoring/report/report.go

```diff
@@ -110,6 +110,12 @@ func getReporterConfig(
 			rc = merged
 		}
 
+		format, err := monitoringConfig.String("format", -1)
+		if err != nil {
+			return "", nil, err
+		}
+		rc.SetString("parameters.format", -1, format)
```

Comment:
uff. I wish we could make this 'cleaner'.

Why have this config check here? The evaluation of the `format` setting is better handled in the elasticsearch reporter, as it's a local 'feature' to the ES reporter only.

Reply:
Yeah, I'm not a big fan of this either, but the format depends on whether the user is setting `xpack.monitoring.*` or `monitoring.*`. This code here coupled with https://github.com/elastic/beats/pull/9260/files#diff-f7226280e7734d6e3d88e7dde232602eR1010 is letting us determine the format based on the configuration and passing it down to the reporter(s).

I couldn't come up with a nicer way to pass down to the ES reporter whether the user had set `xpack.monitoring.*` or `monitoring.*`. Any ideas?

Reply:
As we dynamically build a config object here, we can't do that much. We still need to forward the `_format` setting somehow. This can be done by either:
1. extract the configuration and pass it as extra parameter to the reporters constructor
2. write a setting in the constructed configuration object

You did opt for solution number 2. None is really nice/clean here, so I'm fine with this chosen approach. What feels somewhat more hacky about it is that you overwrite/add to the `parameters` setting, which is already a setting of the elasticsearch reporter, used to set additional HTTP parameters. I think I would prefer the reporter itself to interpret `_format` the way it is needed. That is: replace `rc.SetString("parameters._format", -1, _format)` with `rc.SetString("_format", -1, _format)`, and have the reporter read and apply the `_format` setting as required. E.g. have the ES reporter read the `_format` setting and add it internally to it's parameters. This way it's always the same setting we're dealing with, pushing the semantic (how to use it) to the reporter, not anywhere inbetween. This way `_format` becomes less intermingled with user-provided settings.

Reply:
Oh, it wasn't obvious to me that the `parameters` setting was meant just for HTTP parameters, but I see that now. Thanks for drawing my attention to this.

I agree then that we shouldn't corrupt that setting like I am doing at the moment. Will fix per your recommendation.